Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.cpp

```diff
@@ -70,6 +70,9 @@ void WebDataListSuggestionPicker::displayWithActivationType(WebCore::DataListSug
         return;
     }
 
+    if (type == WebCore::DataListSuggestionActivationType::ShowPickerCalled)
+        close();
```

Comment:
Why is this needed?

Reply:
You need to close the list before showing it again else it ends up toggling and you end up with a closed datalist if you press the trigger button twice.

Reply:
Does that answer the question well enough or would you like me to dig into the exact reason @pxlcoder ?

Reply:
I think it might be good to look a bit more into that, since this is a little confusing, and seems like a workaround.

It may have something to do with the fact that we close the list when clicking away from it?

Reply:
Yeah it's definitely a workaround I'm not entirely sure on the cause of the issue it's working around.

If it was clicking outside I'm not sure why closing and opening it would fix it.

I'll build it locally again and try and track down the exact code path it goes down and see if I can be fixed a slightly more proper way.

Reply:
I've just realised it's this https://bugs.webkit.org/show_bug.cgi?id=261639 the datalist is rendering just in the wrong location. When I do it on my other monitor it works fine. So I'll go ahead and remove this "fix" and it'll just be part of that bug to fix it.